test/DynamoCoreWpfTests/ViewExtensionTests.cs

```diff
@@ -62,6 +84,64 @@ public void Loaded(ViewLoadedParams p)
             {
                 Counter++;
             };
+
+            var window = new Window
+            {
+                // Set the owner of the window to tuIDhe Dynamo window.
+                Owner = p.DynamoWindow
+            };
+
+            p.AddToExtensionsSideBar(this, window);
+        }
+
+        public void Shutdown()
+        {
+
+        }
+
+        public void Dispose()
+        {
+
+        }
+    }
+
+    // A sample view extension which will be added to the extensions side bar
+    public class ExtensionsSideBarViewExtension : IViewExtension
+    {
+        public int Counter = 0;
+        public string uID = "ExtensionsSideBarDummyID";
+
+        public string UniqueId
+        {
+            set { uID = value; }
```

Comment:
Added the setter function here only to this class because we need another instance(with a different UID) of the same extension. The extension manager checks for duplicate UID while adding extensions and it wont even call the loaded function.

Reply:
I don't think this will have any issues - you're only modifying your test extension right? Does this compile?

Reply:
Yes, it is compiled and the test runs fine.